airbyte-server/src/main/java/io/airbyte/server/ApplicationShutdownListener.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.server;
+
+import io.airbyte.commons.server.handlers.HealthCheckHandler;
+import io.micronaut.context.event.ApplicationEventListener;
+import io.micronaut.runtime.event.ApplicationShutdownEvent;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import java.lang.invoke.MethodHandles;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ApplicationShutdownListener sets the ready flag on the HealthCheckHandler to false when it
+ * receives a shutdown event.
+ */
+@Singleton
+public class ApplicationShutdownListener implements ApplicationEventListener<ApplicationShutdownEvent> {
+
+  // TODO: log4j is likely unusable during shutdown; so logs will never show up. Need some kind of
+  // failsafe logger here
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  @Inject
+  private HealthCheckHandler healthCheckHandler;
+
+  @Override
+  public void onApplicationEvent(ApplicationShutdownEvent event) {
```

Comment:
@justenwalker It probably makes more sense to just have the `HealthCheckHandler` implement the `ApplicationEventListener<ApplicationShutdownEvent>` so it can handle the event directly and modify the flag.  No need to have a separate singleton to do this here.

Reply:
I tried this first, but you can only implement a generic interface once, due to type erasure. I can either implement `ApplicationEventListener<ServiceReadyEvent>` or `ApplicationEventListener<ApplicationShutdownEvent>` on `HealthCheckHandler`; but not both